### PR TITLE
ReplaceHooks method for logger

### PR DESCRIPTION
**Pull request by betrok**
Safe way to modify hooks. Related issues:  #715, #701

### CODE REVIEW

File: logger.go [hunk 1]
```diff
@@ -336,3 +336,9 @@ func (logger *Logger) AddHook(hook Hook) {
 	defer logger.mu.Unlock()
 	logger.Hooks.Add(hook)
 }
+
+func (logger *Logger) ReplaceHooks(hooks LevelHooks) {
```
Comment (dgsb) [inline logger.go]: Can we have the former hooks returned from this method ? It could practical in order to restore the former hooks.

**Comment by dgsb** [comment]
Hello, thanks for your contribution. Can you have a look at the slight modification I have requested ?

**Comment by betrok** [comment]
Ok, i added the return value.

**Comment by dgsb** [comment]
@betrok thanks for your contribution

**Comment by XieWeiXie** [comment]
When dealing with a large volume of logs, the ReplaceHook may cause CPU spikes.
image

**Comment by betrok** [comment]
It has nothing to do with this pool request, check what your hooks do.

**Comment by XieWeiXie** [comment]
> It has nothing to do with this pool request, check what your hooks do.

what should i do?

i want to change some filed in logger message , \r \n \t and so on。

this is my example hook。

type ReplaceHook struct {
	Old []string
	New string
}

func (hook *ReplaceHook) Fire(entry *logrus.Entry) error {
	var newMessage = entry.Message
	for _, i := range hook.Old {
		newMessage = strings.ReplaceAll(newMessage, i, hook.New)
	}
	entry.Message = newMessage
	return nil
}

func (hook *ReplaceHook) Levels() []logrus.Level {
	return logrus.AllLevels
}


it worked , but it caused cpu higher than no hook。

**Pull request merged**